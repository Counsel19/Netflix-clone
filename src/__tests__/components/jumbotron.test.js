/* eslint-disable testing-library/no-node-access */
import React from "react";
import { render, screen } from "@testing-library/react";
import { Jumbotron } from "../../components";
import jumboData from "../../fixtures/jumbo.json";

describe("<Jumbotron />", () => {
  it("renders the <Jumbotron /> with populated data", () => {
    const { container } = render(
      <Jumbotron.Container>
        {jumboData.map((item) => {
          return (
            <Jumbotron key={item.id} direction={item.direction}>
              <Jumbotron.Pane>
                <Jumbotron.Title>{item.title}</Jumbotron.Title>
                <Jumbotron.SubTitle>{item.subTitle}</Jumbotron.SubTitle>
              </Jumbotron.Pane>

              <Jumbotron.Pane>
                <Jumbotron.Image src={item.image} data-testid="jumbotron-image"/>
              </Jumbotron.Pane>
            </Jumbotron>
          );
        })}
      </Jumbotron.Container>
    );

    expect(screen.getByText('Enjoy on your TV.')).toBeTruthy();
    expect(screen.getByText('Watch on smart TVs, PlayStation, Xbox, Chromecast, Apple TV, Blu-ray players and more.')).toBeTruthy();
    expect(screen.getAllByTestId('jumbotron-image')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot()
  });

  it("renders the <Jumbotron /> with populated data but without a direction", () => {
    const { container } = render(
      <Jumbotron.Container>
        {jumboData.map((item) => {
          return (
            <Jumbotron key={item.id} data-testid={`${item.id}-jumbotron-default-direction`}>
              <Jumbotron.Pane>
                <Jumbotron.Title>{item.title}</Jumbotron.Title>
                <Jumbotron.SubTitle>{item.subTitle}</Jumbotron.SubTitle>
              </Jumbotron.Pane>

              <Jumbotron.Pane>
                <Jumbotron.Image src={item.image} data-testid="jumbotron-image"/>
              </Jumbotron.Pane>
            </Jumbotron>
          );
        })}
      </Jumbotron.Container>
    );
        expect(screen.getByTestId('1-jumbotron-default-direction')).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot()
  });

});
