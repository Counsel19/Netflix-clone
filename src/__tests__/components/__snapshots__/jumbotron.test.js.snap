// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Jumbotron /> renders the <Jumbotron /> with populated data 1`] = `
<div
  class="jumbotron__Container-sc-tihwqv-6 dPArq"
>
  <div
    class="jumbotron__Item-sc-tihwqv-0 dLfgbx"
  >
    <div
      class="jumbotron__Inner-sc-tihwqv-1 fNdoXS"
      direction="row"
    >
      <div
        class="jumbotron__Pane-sc-tihwqv-2 KvzUU"
      >
        <h1
          class="jumbotron__Title-sc-tihwqv-3 bKXkoU"
        >
          Enjoy on your TV.
        </h1>
        <h2
          class="jumbotron__SubTitle-sc-tihwqv-4 EboBR"
        >
          Watch on smart TVs, PlayStation, Xbox, Chromecast, Apple TV, Blu-ray players and more.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-sc-tihwqv-2 KvzUU"
      >
        <img
          class="jumbotron__Image-sc-tihwqv-5 ksyTgQ"
          data-testid="jumbotron-image"
          src="/images/misc/home-tv.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="jumbotron__Item-sc-tihwqv-0 dLfgbx"
  >
    <div
      class="jumbotron__Inner-sc-tihwqv-1 dJsRZb"
      direction="row-reverse"
    >
      <div
        class="jumbotron__Pane-sc-tihwqv-2 KvzUU"
      >
        <h1
          class="jumbotron__Title-sc-tihwqv-3 bKXkoU"
        >
          Download your programmes to watch on the go.
        </h1>
        <h2
          class="jumbotron__SubTitle-sc-tihwqv-4 EboBR"
        >
          Save your data and watch all your favourites offline.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-sc-tihwqv-2 KvzUU"
      >
        <img
          class="jumbotron__Image-sc-tihwqv-5 ksyTgQ"
          data-testid="jumbotron-image"
          src="/images/misc/home-mobile.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="jumbotron__Item-sc-tihwqv-0 dLfgbx"
  >
    <div
      class="jumbotron__Inner-sc-tihwqv-1 fNdoXS"
      direction="row"
    >
      <div
        class="jumbotron__Pane-sc-tihwqv-2 KvzUU"
      >
        <h1
          class="jumbotron__Title-sc-tihwqv-3 bKXkoU"
        >
          Watch everywhere.
        </h1>
        <h2
          class="jumbotron__SubTitle-sc-tihwqv-4 EboBR"
        >
          Stream unlimited films and TV programmes on your phone, tablet, laptop and TV without paying more.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-sc-tihwqv-2 KvzUU"
      >
        <img
          class="jumbotron__Image-sc-tihwqv-5 ksyTgQ"
          data-testid="jumbotron-image"
          src="/images/misc/home-imac.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Jumbotron /> renders the <Jumbotron /> with populated data but without a direction 1`] = `
<div
  class="jumbotron__Container-sc-tihwqv-6 dPArq"
>
  <div
    class="jumbotron__Item-sc-tihwqv-0 dLfgbx"
    data-testid="1-jumbotron-default-direction"
  >
    <div
      class="jumbotron__Inner-sc-tihwqv-1 fNdoXS"
      direction="row"
    >
      <div
        class="jumbotron__Pane-sc-tihwqv-2 KvzUU"
      >
        <h1
          class="jumbotron__Title-sc-tihwqv-3 bKXkoU"
        >
          Enjoy on your TV.
        </h1>
        <h2
          class="jumbotron__SubTitle-sc-tihwqv-4 EboBR"
        >
          Watch on smart TVs, PlayStation, Xbox, Chromecast, Apple TV, Blu-ray players and more.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-sc-tihwqv-2 KvzUU"
      >
        <img
          class="jumbotron__Image-sc-tihwqv-5 ksyTgQ"
          data-testid="jumbotron-image"
          src="/images/misc/home-tv.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="jumbotron__Item-sc-tihwqv-0 dLfgbx"
    data-testid="2-jumbotron-default-direction"
  >
    <div
      class="jumbotron__Inner-sc-tihwqv-1 fNdoXS"
      direction="row"
    >
      <div
        class="jumbotron__Pane-sc-tihwqv-2 KvzUU"
      >
        <h1
          class="jumbotron__Title-sc-tihwqv-3 bKXkoU"
        >
          Download your programmes to watch on the go.
        </h1>
        <h2
          class="jumbotron__SubTitle-sc-tihwqv-4 EboBR"
        >
          Save your data and watch all your favourites offline.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-sc-tihwqv-2 KvzUU"
      >
        <img
          class="jumbotron__Image-sc-tihwqv-5 ksyTgQ"
          data-testid="jumbotron-image"
          src="/images/misc/home-mobile.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="jumbotron__Item-sc-tihwqv-0 dLfgbx"
    data-testid="3-jumbotron-default-direction"
  >
    <div
      class="jumbotron__Inner-sc-tihwqv-1 fNdoXS"
      direction="row"
    >
      <div
        class="jumbotron__Pane-sc-tihwqv-2 KvzUU"
      >
        <h1
          class="jumbotron__Title-sc-tihwqv-3 bKXkoU"
        >
          Watch everywhere.
        </h1>
        <h2
          class="jumbotron__SubTitle-sc-tihwqv-4 EboBR"
        >
          Stream unlimited films and TV programmes on your phone, tablet, laptop and TV without paying more.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-sc-tihwqv-2 KvzUU"
      >
        <img
          class="jumbotron__Image-sc-tihwqv-5 ksyTgQ"
          data-testid="jumbotron-image"
          src="/images/misc/home-imac.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;
